teps/0112-replace-volumes-with-workspaces.md

```diff
@@ -256,25 +268,186 @@ spec:
 
 Pros:
 - Flexible, not tied to Kubernetes
+- Workspace "types" could be leveraged in the future to label workspaces as "input" and "output", to address
+  [workspace dependencies](./0063-workspace-dependencies.md). (This is also a potential downside, as it signals
+  that "type" is too generic.)
 
 Cons:
 - `Task` authors may want to use other types of Volumes as part of Task implementation, such as hostPath volumes
 
-### Support for "dynamic" Workspaces
+#### Allow running all Steps in the same container
 
-In order to meet the use case of [dynamically created volumes](#dynamically-created-volumes), we could allow the user to specify
-that they don't want to validate the existence of Secrets or ConfigMaps used in `PipelineRun` Workspace bindings until any `TaskRuns`
-that use these Workspaces are created. For example:
+In this solution, we could allow multiple Steps to run in the same container. For example:
 
 ```yaml
 apiVersion: tekton.dev/v1beta1
+kind: Task
+spec:
+  steps:
+  - name: step1
+    image: ubuntu
+    script: echo "hello"
+  - name: step2
+    script: echo "goodbye"
+```
+
+In this example, all Steps would run in a container running the "ubuntu" image.
+(Alternatively, `image` could be a field under `task.spec` instead of `task.spec.steps`.)
+
+Pros:
+- Other CI/CD systems have "Step" equivalents that share a filesystem, including
+  [Github Actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idsteps) 
+  and [CircleCI](https://circleci.com/docs/jobs-steps#steps-overview). (By itself, this isn't a good enough reason
+  to implement something, but it shows that it is a workable solution for many use cases.)
+
+Cons:
+- This wouldn't address use cases where you want Steps to run in separate containers, such as for [Hermekton](https://github.com/tektoncd/community/blob/main/teps/0025-hermekton.md).
+- This wouldn't help with sharing volumes between Steps and Sidecars, since Sidecars often run different images than Steps.
+
+#### Allow Tasks to declare "WorkingDirectories"
```

Comment:
Same question as above. In addition, it is already possible through `stepTemplate`.